pkg/thumbnails/storage/filesystem.go

```diff
@@ -0,0 +1,80 @@
+package storage
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/owncloud/ocis-thumbnails/pkg/config"
+)
+
+// NewFileSystemStorage creates a new instanz of FileSystem
+func NewFileSystemStorage(cfg config.FileSystemStorage) FileSystem {
+	return FileSystem{
+		dir: cfg.RootDirectory,
+	}
+}
+
+// FileSystem represents a storage for the thumbnails using the local file system.
+type FileSystem struct {
+	dir string
+}
+
+// Get loads the image from the file system.
+func (s FileSystem) Get(key string) []byte {
+	content, err := ioutil.ReadFile(filepath.Join(s.dir, key))
+	if err != nil {
+		return nil
+	}
+
+	return content
+}
+
+// Set writes the image to the file system.
+func (s FileSystem) Set(key string, img []byte) error {
+	path := filepath.Join(s.dir, key)
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0700); err != nil {
```

Comment:
does it need execution permissions? or can this be set to `0600`?

Reply:
Yes, execution permission is needed to be able to read or write files in the directory.